addons/mail/static/src/component_hooks/use_component_to_model/use_component_to_model.js

```diff
@@ -17,23 +17,24 @@ const { onWillUpdateProps, useComponent } = owl.hooks;
  */
 export function useComponentToModel({ fieldName, modelName, propNameAsRecordLocalId }) {
     const component = useComponent();
-    const record = component.env.models[modelName].get(component.props[propNameAsRecordLocalId]);
+    const { modelManager } = component.env.services.messaging;
+    const record = modelManager.models[modelName].get(component.props[propNameAsRecordLocalId]);
     if (record) {
         record.update({ [fieldName]: component });
     }
     onWillUpdateProps(nextProps => {
-        const currentRecord = component.env.models[modelName].get(component.props[propNameAsRecordLocalId]);
+        const currentRecord = modelManager.models[modelName].get(component.props[propNameAsRecordLocalId]);
         if (currentRecord) {
             currentRecord.update({ [fieldName]: clear() });
         }
-        const nextRecord = component.env.models[modelName].get(nextProps[propNameAsRecordLocalId]);
+        const nextRecord = modelManager.models[modelName].get(nextProps[propNameAsRecordLocalId]);
         if (nextRecord) {
             nextRecord.update({ [fieldName]: component });
         }
     });
     const __destroy = component.__destroy;
     component.__destroy = parent => {
-        const record = component.env.models[modelName].get(component.props[propNameAsRecordLocalId]);
+        const record = modelManager.models[modelName].get(component.props[propNameAsRecordLocalId]);
```

Comment:
Hm... would be better that awareness of a model manager stays in the internal code of model.
Would be nicer just using messaging record to access from models.
Maybe there's an issue when the messaging record is deleted?

It's fine for this hook to handle cases that `services.messaging.get()` returns `undefined`.
I suggested in comment before to turn `modelManager.getMessaging()` as just `services.messaging.get()` but the async behaviour looks undesirable here.
I feel the proper way is to split `getMessaging()` inner-code with the awaiting initialize/creation, and have another piece of code to get messaging record, which is synchronous.

Reply:
I agree to avoid leaking `modelManager`, but I think it's out of scope here, for all intents and purposes ours hooks should be considered as part of internal code too.
The end goal would be to get rid of the hooks and integrate them to messaging component registry, which itself should be integrated to models/model manager.